src/java.base/share/classes/java/lang/invoke/VarForm.java

```diff
@@ -93,14 +109,39 @@ final MethodType getMethodType(int type) {
 
     @ForceInline
     final MemberName getMemberName(int mode) {
-        // TODO calculate lazily
         MemberName mn = memberName_table[mode];
         if (mn == null) {
-            throw new UnsupportedOperationException();
+            mn = resolveMemberName(mode, false);
+        }
+        return mn;
+    }
+
+    @ForceInline
+    final MemberName getMemberNameOrNull(int mode) {
+        MemberName mn = memberName_table[mode];
+        if (mn == null) {
+            mn = resolveMemberName(mode, true);
         }
         return mn;
     }
 
+    @DontInline
+    MemberName resolveMemberName(int mode, boolean graceful) {
+        AccessMode value = AccessMode.values()[mode];
+        String methodName = value.methodName();
+        try {
+            MethodType type = methodType_table[value.at.ordinal()].insertParameterTypes(0, VarHandle.class);
+            return memberName_table[mode]
+                = MethodHandles.Lookup.IMPL_LOOKUP
+                    .resolveOrFail(REF_invokeStatic, implClass, methodName, type);
```

Comment:
An alternative way can be:

```
final MemberName getMemberName(int mode) {
     MemberName mn = getMemberNameOrNull(mode);
     if (mn == null) {
           throw new UnsupportedOperationException();
     }
 }       
```

This way `resolveMemberName(int mode)` can simply call IMPL_LOOKUP.resolveOrNull`.